test/functional/feature_taproot.py

```diff
@@ -1451,7 +1451,7 @@ def run_test(self):
 
         # Pre-taproot activation tests.
         self.log.info("Pre-activation tests...")
-        self.test_spenders(self.nodes[0], spenders_taproot_inactive(), input_counts=[1, 2, 2, 2, 2, 3])
+        self.test_spenders(self.nodes[0], spenders_taproot_inactive() + spenders_taproot_inactive() + spenders_taproot_inactive(), input_counts=[1, 2, 2, 2, 2, 3])
```

Comment:
I don't understand what this change is doing.

Reply:
It's running every test twice.

What motivated this: before this change, there are only 10 tests, the majority of which were non-standard. As a transaction is only standard when (at least) all its inputs are, this meant that almost all test transactions were expected to be nonstandard (as they contain more than one input). This means not much is tested: having one test incorrectly marked as standard would often not be caught, as it'd get combined with a nonstandard one. Having multiple instances of every test improves that somewhat.

Reply:
ok, in that case I think this would be a bit more obvious:

```diff
diff --git a/test/functional/feature_taproot.py b/test/functional/feature_taproot.py
index ca523572bc..5984ea4161 100755
--- a/test/functional/feature_taproot.py
+++ b/test/functional/feature_taproot.py
@@ -1143,6 +1143,9 @@ def spenders_taproot_inactive():
     add_spender(spenders, "inactive/scriptpath_valid_opsuccess", key=sec, tap=tap, leaf="op_success", standard=False, inputs=[getter("sign")])
     add_spender(spenders, "inactive/scriptpath_valid_opsuccess", key=sec, tap=tap, leaf="op_success", standard=False, inputs=[getter("sign")], sighash=bitflipper(default_sighash))
 
+    # Repeat each test 3 times so that any input which is non-standard is more likely to be caught
+    spenders *= 3
+
     return spenders
 
 # Consensus validation flags to use in dumps for tests with "legacy/" or "inactive/" prefix.
@@ -1451,7 +1454,7 @@ class TaprootTest(BitcoinTestFramework):
 
         # Pre-taproot activation tests.
         self.log.info("Pre-activation tests...")
-        self.test_spenders(self.nodes[0], spenders_taproot_inactive() + spenders_taproot_inactive() + spenders_taproot_inactive(), input_counts=[1, 2, 2, 2, 2, 3])
+        self.test_spenders(self.nodes[0], spenders_taproot_inactive(), input_counts=[1, 2, 2, 2, 2, 3])
 
 
```

Reply:
I've changed the approach, and moved it to a different commit.